{
  "affected_release": null,
  "package_state": [
    {
      "product_name": "Red Hat Enterprise Linux 6",
      "fix_state": "Out of support scope",
      "package_name": "kernel",
      "cpe": "cpe:/o:redhat:enterprise_linux:6"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Out of support scope",
      "package_name": "kernel",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Out of support scope",
      "package_name": "kernel-rt",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Will not fix",
      "package_name": "kernel",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Will not fix",
      "package_name": "kernel-rt",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 9",
      "fix_state": "Affected",
      "package_name": "kernel",
      "cpe": "cpe:/o:redhat:enterprise_linux:9"
    },
    {
      "product_name": "Red Hat Enterprise Linux 9",
      "fix_state": "Affected",
      "package_name": "kernel-rt",
      "cpe": "cpe:/o:redhat:enterprise_linux:9"
    }
  ],
  "threat_severity": "Moderate",
  "public_date": "2024-08-22T00:00:00Z",
  "bugzilla": {
    "description": "kernel: blktrace: fix use after free for struct blk_trace",
    "id": "2307166",
    "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2307166"
  },
  "cvss": {
    "cvss_base_score": "",
    "cvss_scoring_vector": "",
    "status": ""
  },
  "cvss3": {
    "cvss3_base_score": "6.7",
    "cvss3_scoring_vector": "CVSS:3.1/AV:L/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H",
    "status": "draft"
  },
  "iava": "",
  "cwe": "CWE-416",
  "statement": "",
  "acknowledgement": "",
  "name": "CVE-2022-48913",
  "document_distribution": "",
  "details": [
    "In the Linux kernel, the following vulnerability has been resolved:\nblktrace: fix use after free for struct blk_trace\nWhen tracing the whole disk, 'dropped' and 'msg' will be created\nunder 'q-\u003edebugfs_dir' and 'bt-\u003edir' is NULL, thus blk_trace_free()\nwon't remove those files. What's worse, the following UAF can be\ntriggered because of accessing stale 'dropped' and 'msg':\n==================================================================\nBUG: KASAN: use-after-free in blk_dropped_read+0x89/0x100\nRead of size 4 at addr ffff88816912f3d8 by task blktrace/1188\nCPU: 27 PID: 1188 Comm: blktrace Not tainted 5.17.0-rc4-next-20220217+ #469\nHardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS ?-20190727_073836-4\nCall Trace:\n\u003cTASK\u003e\ndump_stack_lvl+0x34/0x44\nprint_address_description.constprop.0.cold+0xab/0x381\n? blk_dropped_read+0x89/0x100\n? blk_dropped_read+0x89/0x100\nkasan_report.cold+0x83/0xdf\n? blk_dropped_read+0x89/0x100\nkasan_check_range+0x140/0x1b0\nblk_dropped_read+0x89/0x100\n? blk_create_buf_file_callback+0x20/0x20\n? kmem_cache_free+0xa1/0x500\n? do_sys_openat2+0x258/0x460\nfull_proxy_read+0x8f/0xc0\nvfs_read+0xc6/0x260\nksys_read+0xb9/0x150\n? vfs_write+0x3d0/0x3d0\n? fpregs_assert_state_consistent+0x55/0x60\n? exit_to_user_mode_prepare+0x39/0x1e0\ndo_syscall_64+0x35/0x80\nentry_SYSCALL_64_after_hwframe+0x44/0xae\nRIP: 0033:0x7fbc080d92fd\nCode: ce 20 00 00 75 10 b8 00 00 00 00 0f 05 48 3d 01 f0 ff ff 73 31 c3 48 83 1\nRSP: 002b:00007fbb95ff9cb0 EFLAGS: 00000293 ORIG_RAX: 0000000000000000\nRAX: ffffffffffffffda RBX: 00007fbb95ff9dc0 RCX: 00007fbc080d92fd\nRDX: 0000000000000100 RSI: 00007fbb95ff9cc0 RDI: 0000000000000045\nRBP: 0000000000000045 R08: 0000000000406299 R09: 00000000fffffffd\nR10: 000000000153afa0 R11: 0000000000000293 R12: 00007fbb780008c0\nR13: 00007fbb78000938 R14: 0000000000608b30 R15: 00007fbb780029c8\n\u003c/TASK\u003e\nAllocated by task 1050:\nkasan_save_stack+0x1e/0x40\n__kasan_kmalloc+0x81/0xa0\ndo_blk_trace_setup+0xcb/0x410\n__blk_trace_setup+0xac/0x130\nblk_trace_ioctl+0xe9/0x1c0\nblkdev_ioctl+0xf1/0x390\n__x64_sys_ioctl+0xa5/0xe0\ndo_syscall_64+0x35/0x80\nentry_SYSCALL_64_after_hwframe+0x44/0xae\nFreed by task 1050:\nkasan_save_stack+0x1e/0x40\nkasan_set_track+0x21/0x30\nkasan_set_free_info+0x20/0x30\n__kasan_slab_free+0x103/0x180\nkfree+0x9a/0x4c0\n__blk_trace_remove+0x53/0x70\nblk_trace_ioctl+0x199/0x1c0\nblkdev_common_ioctl+0x5e9/0xb30\nblkdev_ioctl+0x1a5/0x390\n__x64_sys_ioctl+0xa5/0xe0\ndo_syscall_64+0x35/0x80\nentry_SYSCALL_64_after_hwframe+0x44/0xae\nThe buggy address belongs to the object at ffff88816912f380\nwhich belongs to the cache kmalloc-96 of size 96\nThe buggy address is located 88 bytes inside of\n96-byte region [ffff88816912f380, ffff88816912f3e0)\nThe buggy address belongs to the page:\npage:000000009a1b4e7c refcount:1 mapcount:0 mapping:0000000000000000 index:0x0f\nflags: 0x17ffffc0000200(slab|node=0|zone=2|lastcpupid=0x1fffff)\nraw: 0017ffffc0000200 ffffea00044f1100 dead000000000002 ffff88810004c780\nraw: 0000000000000000 0000000000200020 00000001ffffffff 0000000000000000\npage dumped because: kasan: bad access detected\nMemory state around the buggy address:\nffff88816912f280: fa fb fb fb fb fb fb fb fb fb fb fb fc fc fc fc\nffff88816912f300: fa fb fb fb fb fb fb fb fb fb fb fb fc fc fc fc\n\u003effff88816912f380: fa fb fb fb fb fb fb fb fb fb fb fb fc fc fc fc\n^\nffff88816912f400: fa fb fb fb fb fb fb fb fb fb fb fb fc fc fc fc\nffff88816912f480: fa fb fb fb fb fb fb fb fb fb fb fb fc fc fc fc\n=================================================================="
  ],
  "references": [
    "https://www.cve.org/CVERecord?id=CVE-2022-48913\nhttps://nvd.nist.gov/vuln/detail/CVE-2022-48913\nhttps://lore.kernel.org/linux-cve-announce/2024082215-CVE-2022-48913-94d0@gregkh/T"
  ]
}